### PR TITLE
santa: add process.entity_id constructed from agent.id, pid and pidversion

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds a `process.entity_id` from the agent.id, the process.pid and the process.pidversion (stored in santa.pidversion).

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates https://github.com/elastic/integrations/pull/3347#pullrequestreview-974633387 and https://github.com/elastic/integrations/pull/3347#pullrequestreview-975997434.

### CODE REVIEW

File: packages/santa/changelog.yml [hunk 1]
```diff
@@ -1,8 +1,13 @@
 # newer versions go on top
+- version: "3.1.0"
+  changes:
+    - description: Add `process.entity_id` field.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/1
```
Comment (efd6) [inline packages/santa/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/3373
```

File: packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -16,6 +16,14 @@ processors:
...
       pattern_definitions:
         NOT_SEPARATOR: '[^\|]+'
+  - set:
+      field: process.entity_id
+      value: "{{{process.pid}}}-{{{santa.pidversion}}}"
+      if: "ctx.process?.pid != null && ctx.santa?.pidversion != null"
+  - set:
+      field: process.entity_id
+      value: "{{{agent.id}}}-{{{process.entity_id}}}"
+      if: "ctx.agent?.id != null && ctx.process?.entity_id != null"
```
Comment (r00tu53r) [inline packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: I am not sure who would set `process.entity_id` before ? Would this condition ever be true ?

**Review by r00tu53r — COMMENTED** [review]
LGTM. Just a minor comment/question.

File: packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (efd6) [reply packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: The previous set processor may have. This part appends the agent.id if it's available.

Comment (r00tu53r) [reply packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: eeks missed it :) This enriches the existing field.

Isn't it worth having `agent.id` which I presume is almost always available for cases where `process.pid` and `santa.pidversion` are unavailable. That way there is at least one other identifying factor ? But none the less the user could always user `agent.id` directly I suppose. 👍

**Review by r00tu53r — APPROVED** [review]

File: packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (efd6) [reply packages/santa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:26]: Yes, that will be available independently.

**Pull request merged**